lib/optimizely/event/batch_event_processor.rb

```diff
@@ -144,6 +151,16 @@ def run
 
         add_to_batch(item) if item.is_a? Optimizely::UserEvent
       end
+    rescue SignalException
```

Comment:
This seems like an important change, but it isn't discussed in the summary. Could you explain this change? If this isn't related to LogEvent notification, then I think it should be in its own separate PR.

Reply:
Moved changes which are not related to LogEvent into PR: https://github.com/optimizely/ruby-sdk/pull/200.